sw/host/spiflash/spiflash.cc

```diff
@@ -55,23 +67,55 @@ bool GetFileContents(const std::string &filename, std::string *contents) {
   return true;
 }
 
-// Prints help menu.
+/** Print help menu. */
 static void PrintUsage(int argc, char *argv[]) {
   assert(argc >= 1);
   std::cerr << argv[0] << kUsageString << std::endl;
 }
 
-// Parse command line arguments and store results in |options|.
+/*
+ * Extract the vendor and product ID from `device_id` and store the values
+ * in `options`.
+ *
+ * This function may throw exceptions.
+ *
+ * @return true on success.
+ */
+bool ParseDeviceID(const std::string &device_id, SpiFlashOpts *options) {
+  size_t token_pos = device_id.find(':');
+  if (token_pos == std::string::npos) {
+    std::cerr << "Unable to find device separator ':' in --device_id."
+              << std::endl;
+    return false;
+  }
+
+  std::string vendor_id = device_id.substr(/*pos=*/0, /*len=*/token_pos);
+  options->ftdi_options.device_vendor_id =
+      std::stoi(vendor_id, /*pos=*/0, /*base=*/16);
+
+  std::string product_id =
+      device_id.substr(/*pos=*/ token_pos + 1, /*len=*/std::string::npos);
+  options->ftdi_options.device_product_id =
+      std::stoi(product_id, /*pos=*/0, /*base=*/16);
+
+  return true;
+}
+
+/*
+ * Parse command line arguments and store results in `options`.
+ */
 bool ParseArgs(int argc, char **argv, SpiFlashOpts *options) {
   assert(options);
   const struct option long_options[] = {
       {"input", required_argument, nullptr, 'i'},
+      {"dev_id", required_argument, nullptr, 'd'},
+      {"dev_sn", required_argument, nullptr, 'n'},
```

Comment:
Nit: Personally, I'd prefer `-` in arguments over `_`, so `dev-id` and `dev-sn`

Reply:
Done.